voice/voice-ai/x/OpenAIService/OpenAIUtils.swift

```diff
@@ -0,0 +1,50 @@
+
+import Foundation
+
+struct OpenAIUtils {
+    static func limitConversationContext (_ conversaton: [Message], charactersCount: Int) -> [Message] {
+        var filteredConversation: [Message] = []
+        var totalContentLength = 0
+
+        for message in conversaton.reversed() {
+            guard let content = message.content else {
+                continue
+            }
+            
+            if content.count == 0 {
+                continue
+            }
+            
+            if totalContentLength + content.count <= charactersCount {
+                filteredConversation.insert(message, at: 0)
+                totalContentLength += content.count
+                continue
+            }
+            
+            let charsLeft = charactersCount - totalContentLength;
+            if charsLeft > 0 {
+                let length = min(charsLeft, content.count)
+                let trimmedContent = String(content.suffix(length));
```

Comment:
trim to preserve a full sentence? otherwise it might change the context too much